### ISSUE TITLE
Resolution not upscaled when using mask for preservation - replicate demo

### ISSUE THREAD

**Issue by anilsathyan7** [philz1337x/clarity-upscaler#44]
Adding a mask for preserving part of image leads to image not upscaled in output(resolution). Binary mask is of same size of image in input. If mask not provided, resolution is up-scaled properly.

**Comment by immex-tech** [comment]
+1

**Comment by philz1337x** [comment]
Yes, this is intentional because the masked part will not get upscaled, so I downscale the result to the starting resolution, but the quality of the unmasked area is improved afterward.

**Comment by simbrams** [comment]
Same issue, would it be possible to add an option to force returning the upscaled version ? 
And we would handle the composition ourselves with the masked part. :)
